apps/predictions/lib/predictions_pub_sub.ex

```diff
@@ -229,11 +123,15 @@ defmodule Predictions.PredictionsPubSub do
     end)
   end
 
-  @spec send_data({pid, prediction_key()}, t()) :: broadcast_message()
-  defp send_data({pid, route_stop_direction}, %__MODULE__{
-         predictions_by_key: predictions_by_key
-       }) do
-    new_predictions = predictions_for_key(predictions_by_key, route_stop_direction)
+  @spec send_data({pid, String.t()}, state) :: broadcast_message()
+  defp send_data({pid, keys}, %{ets: table}) do
+    new_predictions = predictions_for_keys(table, keys)
     send(pid, {:new_predictions, new_predictions})
   end
+
+  def to_record(prediction) do
+    {prediction.id, prediction.route.id, prediction.stop.id,
+     if(is_integer(prediction.direction_id), do: Integer.to_string(prediction.direction_id)),
```

Comment:
question: does this if need an else?  In the off chance that the `direction_id` is not an integer?

Reply:
These are coming directly from our parsing of the API, so they definitely should be integers by this point. FWIW if there's no `else` defined, the `if` statement in that case won't error out but will default to `nil` (which will work fine for inserting into the ETS table).